### PR TITLE
QC Report branch with test refinement

### PR DESCRIPTION
**Pull request by turbomam**
- #1427 
- #1333

### CODE REVIEW

**Comment by turbomam** [comment]
@brynnz22 @aclum can we just remove the rule and the invalid examples? It looks like it will build then.

**Comment by aclum** [comment]
@turbomam  I will test that now.

**Comment by aclum** [comment]
@turbomam I need some help getting the mixs.yaml conflicts resolved.

**Comment by turbomam** [comment]
You can delete it from the branch. It gets regenerated by make all, if necessary.

Or I can do it in about two hours.

**Comment by aclum** [comment]
@turbomam  the build is failing b/c the example valid data is missing has_output. I thought you were going to add other tests/change how the model was done so this data would pass.

**Comment by turbomam** [comment]
oh yeah. A reasonable solution would be to assert  `recommended: true` instead of `required: true` on `has_output`'s `slot_usage` in `WorkflowExecutionActivity`. I would especially want @scanon or someone else he designates to sign off on that change.

**Comment by aclum** [comment]
@brynnz22 @turbomam  doing the rule the other way is a lot less desirable since qc_status is not currently required. I would much prefer if we could get the linkml rules working properly. @cmungall  can this be a prioritized activity?

**Comment by aclum** [comment]
To get the value_presence rules to work correctly I had to update linkml to 1.6.6, then @pkalita-lbl helped identify an issue where rules don't inherit to subclasses. Patrick will make a ticket for the linkml issue, this should be released in early January. Workaround would be to copy rules to each subclass.  cc @turbomam @brynnz22

**Comment by pkalita-lbl** [comment]
LinkML issue: https://github.com/linkml/linkml/issues/1802

**Review by aclum — APPROVED** [review]

File: src/schema/nmdc.yaml [hunk 1]
```diff
@@ -223,7 +226,7 @@ classes:
       - extraction_method
       - extraction_target
       - input_mass
-      - quality_control_report
+      - volume
```
Comment (turbomam) [inline src/schema/nmdc.yaml:229]: this doesn't' really have anything to do with QC reports does it? I wonder if I let it sneak in?

File: src/schema/nmdc.yaml [hunk 2]
```diff
@@ -2247,6 +2247,44 @@ enums:
       V-bottom conical tube:
       falcon tube:
 
+  FailureWhatEnum:
```
Comment (turbomam) [inline src/schema/nmdc.yaml:2250]: I think we should make a note somewhere about these permissible values that share names with classes. It's especially interesting in the OWL serialization, becasue the bases of the URIs are quite different

File: src/schema/workflow_execution_activity.yaml [hunk 3]
```diff
@@ -44,7 +45,47 @@ classes:
...
+        preconditions:
+          slot_conditions:
+            qc_status:
+              value_presence: ABSENT
+        postconditions:
+          slot_conditions:
+            has_output:
+              required: true
+
+      # - title: qc_status_pass_qc_where_what_slots_not_specified
```
Comment (turbomam) [inline src/schema/workflow_execution_activity.yaml:75]: are we going to leave this in as commented-out code?

File: src/schema/workflow_execution_activity.yaml [hunk 4]
```diff
@@ -44,7 +45,47 @@ classes:
       - part_of
       - type
       - version
-
+      - qc_status
+      - qc_comment
+      - has_failure_categorization
+    rules:
```
Comment (turbomam) [inline src/schema/workflow_execution_activity.yaml:51]: I haven't scrutinized these. if there's example data to exercise all of the conditions that the rules know about, and those examples pass or fail as excepted, then they are OK with me.

File: src/schema/workflow_execution_activity.yaml [hunk 5]
```diff
@@ -65,6 +106,10 @@ classes:
         structured_pattern:
           syntax: "{id_nmdc_prefix}:wf-{id_shoulder}-{id_blade}{id_version}{id_locus}"
           interpolated: true
+      qc_failure_what:
```
Comment (turbomam) [inline src/schema/workflow_execution_activity.yaml:109]: is `multivalued` being asserted for these slots in a `slot_usage`? It's almost always better to assert that in the global definition of the slot. And in general, we wouldn't want a slot to be multi-valued in some cases and single valued in others.

File: src/schema/nmdc.yaml (hunk 1 above)
Comment (aclum) [reply src/schema/nmdc.yaml:229]: @turbomam  I think this is suppose to be here. You can see commits to main from a month or two ago from @anastasiyaprymolenna  and you. See https://github.com/microbiomedata/nmdc-schema/commit/4448e41336bede06541bd3c3d2457aec6314f6ab
the slot for Class Extraction seems to have fallen out but the slot usage is there so I think this slot should be on this class. @anastasiyaprymolenna  please confirm.

File: src/schema/workflow_execution_activity.yaml (hunk 3 above)
Comment (aclum) [reply src/schema/workflow_execution_activity.yaml]: Deleted

**Pull request merged**